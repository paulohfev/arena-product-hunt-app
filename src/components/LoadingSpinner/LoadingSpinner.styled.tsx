import Image from 'next/image'

import styled, { keyframes } from 'styled-components'

const rotate = keyframes`
  100% {
    transform: rotate(360deg);
  }
`

export const SpinnerIcon = styled(Image)`
  animation: ${rotate} 1s linear infinite;
`
