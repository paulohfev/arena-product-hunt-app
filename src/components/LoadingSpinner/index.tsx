import { SpinnerIcon } from './LoadingSpinner.styled'

const LoadingSpinner = () => {
  return <SpinnerIcon src='/icons/loading.png' alt='loading' width={48} height={48} />
}

export default LoadingSpinner
