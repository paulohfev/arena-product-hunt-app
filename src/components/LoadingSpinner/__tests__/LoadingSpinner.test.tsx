import { render } from '@testing-library/react'

import LoadingSpinner from '../index'

describe('LoadingSpinner', () => {
  it('matches snapshot', () => {
    const { container } = render(<LoadingSpinner />)
    expect(container).toMatchSnapshot()
  })
})
