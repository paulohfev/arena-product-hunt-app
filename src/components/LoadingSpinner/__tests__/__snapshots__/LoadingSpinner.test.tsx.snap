// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner matches snapshot 1`] = `
<div>
  <img
    alt="loading"
    class="sc-66b55120-0 bzrstx"
    data-nimg="1"
    decoding="async"
    height="48"
    loading="lazy"
    src="/_next/image?url=%2Ficons%2Floading.png&w=96&q=75"
    srcset="/_next/image?url=%2Ficons%2Floading.png&w=48&q=75 1x, /_next/image?url=%2Ficons%2Floading.png&w=96&q=75 2x"
    style="color: transparent;"
    width="48"
  />
</div>
`;
